src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactoryProvider.java

```diff
@@ -280,7 +281,14 @@ private static int convertToInt(Object value) {
             builder.preparedStatementCacheQueries(convertToInt(preparedStatementCacheQueries));
         }
 
-        Map<String, String> options = connectionFactoryOptions.getValue(OPTIONS);
+        Map<String, String> options;
+        try {
+            options = connectionFactoryOptions.getValue(OPTIONS);
+        } catch (ClassCastException exception) {
```

Comment:
This should be rather `Object options = getValue(…)` and then inspecting whether the options are a string or a map.

Reply:
sure, will update and rebase